server/src/services/library.service.ts

```diff
@@ -503,7 +512,7 @@ export class LibraryService {
         type: assetType,
         originalFileName,
         sidecarPath,
-        isReadOnly: true,
+        isReadOnly: library?.isReadOnly ?? false,
```

Comment:
I think we can just throw an exception above this if the library is missing: that really shouldn't happen.

Reply:
yeah, I'd actually tried to make that change, but it seems there's quite a few existing tests that are relying on this to work with no library. I'm afraid it's not a small change. It's not a bad idea, but probably better to do in a separate PR if we want to do it to keep this one from growing too large

Reply:
Hmm gotcha. We can do that in a later PR if it involves a lot of other changes.